src/lib/transaction_snark/test/util.ml

```diff
@@ -47,13 +47,48 @@ let genesis_state_body =
   in
   compile_time_genesis.data |> Mina_state.Protocol_state.body
 
+let genesis_state_body_hash =
+  Mina_state.Protocol_state.Body.hash genesis_state_body
+
 let init_stack = Pending_coinbase.Stack.empty
 
+let pending_coinbase_state_stack =
+  { Transaction_snark.Pending_coinbase_stack_state.source = init_stack
+  ; target =
+      Pending_coinbase.Stack.push_state genesis_state_body_hash init_stack
+  }
+
 let apply_parties ledger parties =
+  let parties =
+    match parties with
+    | [ ps ] ->
+        [ ( `Pending_coinbase_init_stack init_stack
+          , `Pending_coinbase_of_statement pending_coinbase_state_stack
+          , ps )
+        ]
+    | ps1 :: ps2 :: rest ->
+        let ps1 =
+          ( `Pending_coinbase_init_stack init_stack
+          , `Pending_coinbase_of_statement pending_coinbase_state_stack
+          , ps1 )
+        in
+        let unchanged_stack_state ps =
+          ( `Pending_coinbase_init_stack init_stack
+          , `Pending_coinbase_of_statement
+              { pending_coinbase_state_stack with
+                source = pending_coinbase_state_stack.target
+              }
+          , ps )
+        in
+        let ps2 = unchanged_stack_state ps2 in
+        ps1 :: ps2 :: List.map rest ~f:unchanged_stack_state
+    | [] ->
```

Comment:
I would order these cases by empty/singleton/at-least-two, which I think is more conventional